utils/luci-app-ffwizard-berlin/luasrc/tools/freifunk/assistent/tools.lua

```diff
@@ -102,3 +102,22 @@ end
 function logger(msg)
         sys.exec("logger -t ffwizard -p 5 '"..msg.."'")
 end
+
+
+function getMergedConfig(section)
+-- merge the values from the "freifunk" config and the community config
+  local data = {}
+  for i, config in ipairs({"freifunk", community}) do
+    uci:foreach(config, "defaults",
+      function(s)
+        if s['.name'] == section then
+          for key, val in pairs(s) do
+            if string.sub(key, 1, 1) ~= '.' then
+              data[key] = val
```

Comment:
This looks a bit like duplicated code to https://github.com/freifunk-berlin/firmware-packages/pull/174/files#diff-ebf9c3e54752b0821aa3608756414bb5R97 - if there is a mistake or a future enhancement, one location can break - what do you think?
-  nothing against merge.

looks to me as one could use getMergedConfig to pass the values to uci at the other location.

Reply:
@niccokunzmann  I have force-pushed a new version of the code, without the duplicated code above.  Check it out please.

Reply:
I looked at it. Duplication is removed.
If I merge some things, it is sometimes not clear, what the value is to take.

```
merge({a:1}, {a:2})
```
could be `{a:1}` or `{a:2}`.

For inheritance, the `{a:1}` would be the case, the first element having the highest priority. Yours seem to merge with the last one having the highest priority.
This should be documented as people may build on that behavior. It may however change in the future, because it is not agreed to, thus introducing bugs.

Reply:
I added this comment to the function in tools.lua. I will squash it into the commit later.  I have just started a build on the buildbot with these changes.  Let me know what you think.

```
--configs: an array of strings, each representing a config file.  The order is important since 
--  the first config file is read, then the following.  Any options in the following config files
--  overwrite the values of any previous config files. e.g. {"freifunk", "profile_berlin"}
--sectionType: the section type to merge. e.g. "defaults"
--sectionName: the section to merge. e.g. "olsrd"
```